application/src/main/java/org/opentripplanner/street/model/edge/StreetEdge.java

```diff
@@ -446,8 +443,15 @@ public I18NString getName() {
     return this.name;
   }
 
+  /**
+   * Update the name of the edge after it has been constructed. This method also sets the bogusName
+   * property to false, indicating to the code that maps from edges to steps that this is a real
+   * street name.
+   * @see Edge#hasBogusName()
+   */
   public void setName(I18NString name) {
     this.name = name;
+    this.flags = BitSetUtils.set(flags, HASBOGUSNAME_FLAG_INDEX, false);
```

Comment:
The expression "has bogus name" in the context of an accessor method may imply that a "bogus name" is a thing that may be present or absent, in addition to a "normal name". Seeing these terms in the past I am not sure I realized that "bogusness" was only signalling the provenance of the name, not an extra name. If we want to keep using the term "bogus", is it a better description to say "Name is bogus" instead of "we have a bogus name"?

Taking this a step further, is there a more descriptive term for this than "bogus"? The names are more "inferred", "generic", or "placeholder" than "bogus". Or inverting the boolean condition, "normal" names are those explicitly specified by OSM name tags. Unfortunately "has explicit name from tag" is a little long. NAME_IS_GENERIC or NAME_IS_INFERRED?

Reply:
I fully agree with this but I want to do it in another PR.

I prefer NAME_IS_INFERRED. We can also discuss if the boolean is the right way to model this or if `getName` should return a sealed interface to express the two types of names.